src/quail/Sensor.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+class Sensor;
+
+#include "ad7124-lib/ad7124.h"
+#include <cstdint>
+#include <string>
+
+
+class Sensor
+{
+public:
+    Sensor(uint8_t channel, std::string ch_name, Ad7124::InputSel ainp, Ad7124::InputSel ainm)
+    : channel(channel), ch_name(ch_name), ainp(ainp), ainm(ainm){}; 
+
+    float convertToFloat(uint32_t adc_dataword){return 0.0;}; //over-written by inheritors, returns SI unit value of reading from ADC bin count
+    void initialize(); 
+
+    const uint8_t channel; // ADC channel ID
+    const std::string ch_name; // data channel name
```

Comment:
Does std:string allocate from global heap? I've been trying to avoid using heap, but idk if it's worthwhile. Prevents memory fragmentation on low-mem devices. If this is a static name, can it be static, or in a fixed-size buffer?

Reply:
small enough strings are tossed on the stack (anything less than like 23 chars + a null char) - it can definitely be static, don't think that should be an issue.

Reply:
Wait that's a lie - it can't be static unless we just want to call all pressure ducers "PT" and identify them by their ADC channel (which works, but will result in our first thermocouple being "TC8" or something like that). As is, I do think it will be on the stack, but only if the name is less than X number of chars (actually value of X depends on the STL release, it seems). So not a robust avoid-heap method, but one that works for our little 3-char sensor names (like "PT1").

Reply:
My coding interview at Joby had a fun question about the many meanings of the `static` keyword lol.
Stuff is only static in the sense that's it's not on the heap. Everything in `sys` is static, it's all allocated at boot time. 
We can decide it want to be able to reconfigure the set of sensors after boot, but I would tend to think we don't want to, so we should just have a static elements and an array of references for itteration:

https://github.com/stanford-ssi/SpaceSalmon/blob/af848caf3c54064a4672badc7cd9fe55a89af2cb/src/guppy/System.hpp#L57

Or we can just not and have a heap I'm not super concerned about it.